Shape_detection/include/CGAL/Shape_detection/Region_growing/Polygon_mesh/Plane_face_region.h

```diff
@@ -0,0 +1,462 @@
+    not far from that supporting plane (far being defined by the input distance threshold).
+
+    \tparam GeomTraits
+    a model of `Kernel`
+
+    \tparam PolygonMesh
+    a model of `FaceListGraph`
+
+    \tparam VertexToPointMap
+    a model of `ReadablePropertyMap` whose key type is the vertex type of a polygon mesh and
+    value type is `Kernel::Point_3`
+
+    \cgalModels{RegionType}
+  */
+  template<
+  typename GeomTraits,
+  typename PolygonMesh,
+  typename VertexToPointMap = typename boost::property_map<PolygonMesh, CGAL::vertex_point_t>::const_type>
+  class Plane_face_region {
+
+  public:
+    /// \name Types
+    /// @{
+
+    /// \cond SKIP_IN_MANUAL
+    using Face_graph = PolygonMesh;
+    using Vertex_to_point_map = VertexToPointMap;
+
+    using face_descriptor = typename boost::graph_traits<PolygonMesh>::face_descriptor;
+    using halfedge_descriptor = typename boost::graph_traits<PolygonMesh>::halfedge_descriptor;
+    using vertex_descriptor = typename boost::graph_traits<PolygonMesh>::vertex_descriptor;
+    /// \endcond
+
+    /// Number type.
+    typedef typename GeomTraits::FT FT;
+
+    /// Item type.
+    using Item = face_descriptor;
+    using Region = std::vector<Item>;
+
+    /// Primitive
+    using Primitive = typename GeomTraits::Plane_3;
+
+    /// Region map
+    using Region_index_map = typename boost::property_map<Face_graph, CGAL::dynamic_face_property_t<std::size_t> >::const_type;
+
+    /// @}
+
+  private:
+    using Point_3 = typename GeomTraits::Point_3;
+    using Vector_3 = typename GeomTraits::Vector_3;
+    using Plane_3 = typename GeomTraits::Plane_3;
+    using Triangle_3 = typename GeomTraits::Triangle_3;
+
+    using Squared_length_3 = typename GeomTraits::Compute_squared_length_3;
+    using Squared_distance_3 = typename GeomTraits::Compute_squared_distance_3;
+    using Scalar_product_3 = typename GeomTraits::Compute_scalar_product_3;
+    using Cross_product_3 = typename GeomTraits::Construct_cross_product_vector_3;
+
+  public:
+    /// \name Initialization
+    /// @{
+
+    /*!
+      \brief initializes all internal data structures.
+
+      \tparam NamedParameters
+      a sequence of \ref bgl_namedparameters "Named Parameters"
+
+      \param pmesh
+      an instance of `PolygonMesh` that represents a polygon mesh
+
+      \param np
+      a sequence of \ref bgl_namedparameters "Named Parameters"
+      among the ones listed below
+
+      \cgalNamedParamsBegin
+        \cgalParamNBegin{maximum_distance}
+          \cgalParamDescription{the maximum distance from the furthest vertex of a face to a plane}
+          \cgalParamType{`GeomTraits::FT`}
+          \cgalParamDefault{1}
+        \cgalParamNEnd
+        \cgalParamNBegin{maximum_angle}
+          \cgalParamDescription{the maximum angle in degrees between
+          the normal of a face and the normal of a plane}
+          \cgalParamType{`GeomTraits::FT`}
+          \cgalParamDefault{25 degrees}
```

Comment:
Why 25 ?

Reply:
I think Sebastien copied it just from the Least_square_fit_plane_region. But for this type of primitive as it is normally used for CAD models 25 degrees seems large.

@sloriot You prefer to something smaller? 5?

Reply:
Let's keep the same value as the over region type